tools/scripts/bumpenvs.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+"""
+Update all the BUMPENVS templated lines in the repository based on bumpenvs.yaml.
+
+An example BUMPENVS templated line should look like this:
+
+    # BUMPENVS: - run: docker pull ${tf2_cpu_hashed}
+    - run: docker pull determinedai/environments:py-3.6.9-pytorch-1.4-tf-2.2-cpu-aaa3750
+
+The second line is what will be updated; it will have leading whitespace to match the
+leading whitespace of the template line, and the ${...} in the template will be replaced
+according to the configuraion in bumpenvs.yaml.
+
+Usage: bumpenvs.py path/to/bumpenvs.yaml
+"""
+
+import re
+import string
+import subprocess
+import sys
+import os
+from typing import Dict, List, Tuple
+
+import yaml
+
+magic = "BUMPENVS"
+
+Replacements = List[Tuple[str, str]]
+
+
+def list_bumpenv_paths() -> List[str]:
+    cmd = ["git", "rev-parse", "--show-toplevel"]
+    r = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True, check=True)
+    github_root = r.stdout.strip()
+
+    cmd = ["git", "grep", "-l", magic, github_root]
```

Comment:
non-blocking: Use `-z` and split on `\0`.

Reply:
good idea, done